### PR TITLE
Catch LDAP bind error and search warning log messages in ldap tests

### LINKED ISSUE #2950: LDAP tests produce (expected) console messages
After I added extra logging in LDAP, it now logs messages into CLI when running tests:

```
2019-01-02 12:07:41 [ERROR] LDAP bind failed: InsufficientAccessRightsError: InsufficientAccessRightsError
2019-01-02 12:07:41 [WARN] LDAP Search did not find anything for: eve (0)
```

We should catch these messages so they are not printed (could even add asserts for them).

### PR DESCRIPTION
**Pull request by moundahiwale**
Fixes #2950

### CODE REVIEW

**Comment by xPaw** [comment]
Thanks for the PR!

We should however use stubs like it is done in other tests already, example file: https://github.com/thelounge/thelounge/blob/061fa615cea3297d97daf7d3054933d741c2b302/test/src/command-line/utilsTest.js

That way we will be able to also have a `expect()` call on the log message to test that it actually logs what we want.

File: test/plugins/auth/ldap.js [hunk 1]
```diff
@@ -95,17 +95,27 @@ function testLdapAuth() {
...
 	it("should fail to authenticate with incorrect password", function(done) {
+		const originalLog = log.info;
+
+		log.error = () => {};
+
 		ldapAuth.auth(manager, client, user, wrongPassword, function(valid) {
 			expect(valid).to.equal(false);
 			done();
 		});
+		log.error = originalLog;
```
Comment (xPaw) [inline test/plugins/auth/ldap.js]: You would need to do the restore inside of the callback

**Comment by moundahiwale** [comment]
Tests updated to assert the actual log messages

File: test/plugins/auth/ldap.js [hunk 2]
```diff
@@ -95,15 +97,25 @@ function testLdapAuth() {
...
 		});
 	});
 
 	it("should fail to authenticate with incorrect username", function(done) {
+		let warning = "";
+		stub(log, "warn").callsFake(TestUtil.sanitizeLog((str) => warning += str));
+
 		ldapAuth.auth(manager, client, wrongUser, correctPassword, function(valid) {
 			expect(valid).to.equal(false);
+			expect(warning).to.equal("LDAP Search did not find anything for: eve (0)\n");
```
Comment (xPaw) [inline test/plugins/auth/ldap.js]: @astorije Any particular reason why `sanitizeLog` appends a new line?

**Pull request merged**